### DISCUSSION TITLE
Are there any tools that you recommend to use with the setup?

### DISCUSSION THREAD

**Discussion by cdinu** [UCL-ARC/python-tooling#58] [Q&A]
I am a `requirements.txt` person and try to upgrade myself to use `pyproject.toml`

Looks like everyone says `poetry` is needed to manage `pyproject.toml` projects.
Is this mandatory, optional? (in your experience)

Are there any other utilities I should have when using templates such as this one?

**Comment by paddyroddy** [comment]
There are a million different build tools https://stackoverflow.com/a/64151860/7359333. I've always used `setuptools` but feel free to use whatever you want. Personally, whilst `poetry` seems like it could be good, I've seen enough issues of when `poetry` has "broken" because some package didn't specifically support it - so I have no plans to use it. If you look at the [Python documentation](https://packaging.python.org/en/latest/tutorials/packaging-projects) they show you how to do the same thing in `Hatchling`, `setupools`, `Flit`, `PDM`.
